File: Java/problem896.java
class Solution {
    public boolean isMonotonic(int[] A) {
        if (A[0] < A[A.length - 1]) {
            for (int i = 0; i < A.length - 1; ++i) {
                if (A[i] > A[i + 1]) {
                    return false;
                }
            }
            return true;
        }
        else {
            for (int i = 0; i < A.length - 1; ++i) {
                if (A[i] < A[i + 1]) {
                    return false;
                }
            }
            return true;
        }
    }
}

File: python/problem896.py
class Solution:
    def isMonotonic(self, A: List[int]) -> bool:
        if A[0] < A[-1]:
            for i in range(len(A) - 1):
                if A[i] > A[i + 1]:
                    return False
            return True
        else:
            for i in range(len(A) - 1):
                if A[i] < A[i + 1]:
                    return False
            return True
